### PR TITLE
Make APIs related to scripting and interactive internal for RTM

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by tmat** [comment]
@cston @AnthonyDGreen @jaredpar @ManishJayaswal

File: src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs [hunk 1]
```diff
@@ -740,6 +740,9 @@ private bool Execute(string text)
                 return true;
             }
 
+            private delegate ITypeSymbol GetSubmissionResultTypeDelegate(Compilation compilation, out bool hasValue);
+            private GetSubmissionResultTypeDelegate GetSubmissionResultType = typeof(Compilation).GetTypeInfo().GetDeclaredMethod("GetSubmissionResultType").CreateDelegate<GetSubmissionResultTypeDelegate>();
```
Comment (cston) [inline src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs]: Does this allow tests to pass? If not, consider creating a delegate that returns an error type instead.

Comment (cston) [inline src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs]: Does this allow tests to pass? If not, consider creating a delegate that throws an exception instead.

File: src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs [hunk 2]
```diff
@@ -247,12 +247,14 @@ public CompilationTracker FreezePartialStateWithTree(Solution solution, Document
                     var referencedProject = solution.GetProject(projectReference.ProjectId);
                     if (referencedProject != null)
                     {
+#if SCRIPTING
                         if (referencedProject.IsSubmission)
```
Comment (cston) [inline src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs]: Is this needed? Isn't IsSubmission accessible?

**Comment by cston** [comment]
LGTM

File: src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs (hunk 2 above)
Comment (tmat) [inline src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs]: It is, but WithPreviousSubmission below isn't.

File: src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs (hunk 1 above)
Comment (tmat) [inline src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs]: Yup, it's here to keep most tests running.

**Comment by jaredpar** [comment]
I would prefer removing the `#ifdef` code altogether.  The last two PRs have added two different `#ifdef` patterns to the code base.  That means there are four different ways our code can be compiled and we're testing one of them.

File: src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs (hunk 1 above)
Comment (tmat) [inline src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs]: I'll add a comment.

**Comment by tmat** [comment]
@dotnet-bot Retest this, please.

**Comment by tmat** [comment]
@dotnet-bot Retest this, please.

**Pull request merged**